### PR TITLE
Feature/comporator 281 springboot demo revisited

### PR DESCRIPTION
**Pull request by ghost**
I revisited my homework. ( And i wrote a new logic :D )

### CODE REVIEW

File: Springboot-demo/mvnw.cmd [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+if exist "%USERPROFILE%\mavenrc_post.bat" call "%USERPROFILE%\mavenrc_post.bat"
+if exist "%USERPROFILE%\mavenrc_post.cmd" call "%USERPROFILE%\mavenrc_post.cmd"
+:skipRcPost
+
+@REM pause the script if MAVEN_BATCH_PAUSE is set to 'on'
+if "%MAVEN_BATCH_PAUSE%"=="on" pause
+
+if "%MAVEN_TERMINATE_CMD%"=="on" exit %ERROR_CODE%
+
+cmd /C exit /B %ERROR_CODE%
```
Comment (lyancsie) [inline Springboot-demo/mvnw.cmd]: If you want to use Maven Wrapper, fine - but at least read what it is :)

File: Springboot-demo/src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+spring.application.name=springboot-demo
+spring.datasource.url=jdbc:mysql://localhost:3306/employee
+spring.datasource.username=root
+spring.datasource.password=root
```
Comment (lyancsie) [inline Springboot-demo/src/main/resources/application.properties]: This is a major violation over here. Never load anything with your local passwords or api keys

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/model/Employee.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    @Column
+    private String name;
+    @Column
+    private int age;
+}
+
+
+
+
+
```
Comment (lyancsie) [inline Springboot-demo/src/main/java/com/hollszabolcs/demo/model/Employee.java]: Why so many empty lines?

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/model/Employee.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import jakarta.persistence.*;
+import lombok.*;
+
+@Entity(name = "employees")
+@Data
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@ToString
```
Comment (lyancsie) [inline Springboot-demo/src/main/java/com/hollszabolcs/demo/model/Employee.java]: Read what [@Data](https://projectlombok.org/features/Data) does exactly and you can remove all the redundant annotations.

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@Slf4j
+public class EmployeeController {
+    private final EmployeeService employeeService;
+
+    @PostMapping("/addemployee")
```
Comment (lyancsie) [inline Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java]: in general, `POST/employee` or `POST/employees` is a good api naming. addemployee is kinda weird.

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/repository/EmployeeRepository.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.hollszabolcs.demo.repository;
+
+import com.hollszabolcs.demo.model.Employee;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface EmployeeRepository extends JpaRepository<Employee, Long> {
```
Comment (lyancsie) [inline Springboot-demo/src/main/java/com/hollszabolcs/demo/repository/EmployeeRepository.java]: The id is not a long here, but you are referring here as a long - I am not sure if this project even runs properly

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    Employee addEmployee(@RequestBody Employee employee){
+        log.info("Adding an employee{}",employee);
+        return employeeService.addEmployee(employee);
+    }
```
Comment (lyancsie) [inline Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java:21]: please also create a get endpoint so you can retrieve the data :)

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+public class EmployeeController {
+    private final EmployeeService employeeService;
+
+    @PostMapping("/employee")
+    Employee addEmployee(@RequestBody Employee employee) {
+        log.info("Adding an employee{}", employee);
+        return employeeService.addEmployee(employee);
+    }
+
+    @GetMapping("/employee2")
```
Comment (lyancsie) [inline Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java]: why `employee2`? :)

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/model/Employee.java (hunk 4 above)
Comment (lyancsie) [reply Springboot-demo/src/main/java/com/hollszabolcs/demo/model/Employee.java]: nothing has been changed here

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/service/EmployeeService.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+@RequiredArgsConstructor
+public class EmployeeService {
+    private final EmployeeRepository employeeRepository;
+
+    public Employee addEmployee(Employee employee) {
+        return employeeRepository.save(employee);
+    }
+
+    public List<Employee> findByName(String name){
+        final var employee =employeeRepository.findByName("Tibi");
```
Comment (lyancsie) [inline Springboot-demo/src/main/java/com/hollszabolcs/demo/service/EmployeeService.java]: ?

File: Springboot-demo/src/main/resources/application.properties (hunk 2 above)
Comment (lyancsie) [reply Springboot-demo/src/main/resources/application.properties]: major violation is still here

**Review by lyancsie — CHANGES_REQUESTED** [review]
some major issues are still to be fixed. pls check my comments

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java (hunk 8 above)
Comment (hollszabolcs) [reply Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java]: I forgot to wrote a better name :D

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/service/EmployeeService.java (hunk 9 above)
Comment (hollszabolcs) [reply Springboot-demo/src/main/java/com/hollszabolcs/demo/service/EmployeeService.java]: I was testing my previous idea, and i forgot to delete it

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+public class EmployeeController {
+    private final EmployeeService employeeService;
+
+    @PostMapping("/employee")
+    Employee addEmployee(@RequestBody Employee employee) {
+        log.info("Adding an employee{}", employee);
+        return employeeService.addEmployee(employee);
+    }
+
+    @GetMapping("/get-employee")
```
Comment (lyancsie) [inline Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java]: please get rid of this api naming violation and the empty lines in Employee.java and i will accept it

```suggestion
    @GetMapping("/employees")
```

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+public class EmployeeController {
+    private final EmployeeService employeeService;
+
+    @PostMapping("/employee")
+    Employee addEmployee(@RequestBody Employee employee) {
+        log.info("Adding an employee{}", employee);
+        return employeeService.addEmployee(employee);
+    }
+
+    @GetMapping("/getEmployee")
```
Comment (lyancsie) [inline Springboot-demo/src/main/java/com/hollszabolcs/demo/controller/EmployeeController.java]:
```suggestion
    @GetMapping("/employee")
```

File: Springboot-demo/src/main/java/com/hollszabolcs/demo/model/Employee.java (hunk 3 above)
Comment (lyancsie) [reply Springboot-demo/src/main/java/com/hollszabolcs/demo/model/Employee.java]: still not fixed

**Pull request merged**